### ISSUE TITLE
Invalid core::time:::Instant --fix

### ISSUE THREAD

**Issue by wcampbell0x2a** [rust-lang/rust-clippy#15330]
### Summary

See Reproducer.

### Reproducer

I tried this code:

```sh
$ git clone git@github.com:wcampbell0x2a/backhand.git
$ cargo clippy --fix --allow-dirty -- -D clippy::std_instead_of_core -D clippy::std_instead_of_core -D clippy::alloc_instead_of_core
```

I expected to see this happen:

Instead, this happened:
```
after fixes were automatically applied the compiler reported errors within these files:

  * backhand-cli/src/bin/unsquashfs.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust-clippy/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0432]: unresolved import `core::time::Instant`
  --> backhand-cli/src/bin/unsquashfs.rs:27:28
   |
27 | use core::time::{Duration, Instant};
   |                            ^^^^^^^ no `Instant` in `time`
   |
   = help: consider importing this struct instead:
           std::time::Instant

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0432`.
Original diagnostics will follow.
```

### Version

```text
rustc 1.88.0 (6b00bc388 2025-06-23)
binary: rustc
commit-hash: 6b00bc3880198600130e1cf62b8f8a93494488cc
commit-date: 2025-06-23
host: x86_64-unknown-linux-gnu
release: 1.88.0
LLVM version: 20.1.5
```

**Comment by profetia** [comment]
This should already be fixed in #15165. Have you tried the latest nightly?

**Comment by samueltardieu** [comment]
Confirmed fixed in the nightly version of Clippy. The fix will be included in Rust 1.90.
